ext/mbstring/php_mbregex.c

```diff
@@ -1400,6 +1400,11 @@ PHP_FUNCTION(mb_ereg_search_setpos)
 		return;
 	}
 
+	/* Accept negative position if length of search string can be determined */
+	if ((position < 0) && (!Z_ISUNDEF(MBREX(search_str))) && (Z_TYPE(MBREX(search_str)) == IS_STRING)) {
+		position += Z_STRLEN(MBREX(search_str));
```

Comment:
Also a bit unsure here about the "if length can be determined" business. We could delay computation of the final offset until the search string is known. Not sure whether or not we should.